source/Missions/Services/Agents/Packets/MovementHandler.cs

```diff
@@ -93,23 +101,51 @@ Mission CurrentMission
             }
         }
 
-        private async void PollAgents()
+        private void Handle_Movement(MessagePayload<Movement> payload)
```

Comment:
Create an individual handler for each movement type.

Create a new class for managing movement deltas and possible publishing packets from there (if that class gets too large feel free to separate it)